artiq/dashboard/datasets.py

```diff
@@ -81,6 +82,89 @@ def get_edit_widget(self, initial_value):
     def get_edit_widget_value(self):
         return self.edit_widget.text()
 
+class Creator(QtWidgets.QDialog):
+    def __init__(self, parent, dataset_ctl):
+        QtWidgets.QDialog.__init__(self, parent=parent)
+        self.dataset_ctl = dataset_ctl
+
+        self.setWindowTitle("Create dataset")
+        grid = QtWidgets.QGridLayout()
+        grid.setRowMinimumHeight(1, 35)
+        grid.setColumnMinimumWidth(2, 60)
+        self.setLayout(grid)
+
+        grid.addWidget(QtWidgets.QLabel("Name:"), 0, 0)
+        grid.addWidget(self.key_widget(), 0, 1)
+
+        grid.addWidget(QtWidgets.QLabel("Value:"), 1, 0)
+        grid.addWidget(self.edit_widget(), 1, 1)
+        self.data_type = QtWidgets.QLabel("data type")
+        grid.addWidget(self.data_type, 1, 2)
+        self.value_widget.textChanged.connect(self.dtype)
+
+        grid.addWidget(QtWidgets.QLabel("Persist:"), 2, 0)
+        grid.addWidget(self.persist_widget(), 2, 1)
+
+        self.ok = QtWidgets.QPushButton('&Ok')
+        self.ok.setEnabled(False)
+        self.cancel = QtWidgets.QPushButton('&Cancel')
+        self.buttons = QtWidgets.QDialogButtonBox(self)
+        self.buttons.addButton(self.ok, QtWidgets.QDialogButtonBox.AcceptRole)
+        self.buttons.addButton(self.cancel, QtWidgets.QDialogButtonBox.RejectRole)
+        grid.setRowStretch(3, 1)
+        grid.addWidget(self.buttons, 4, 0, 1, 3)
+        self.buttons.accepted.connect(self.accept)
+        self.buttons.rejected.connect(self.reject)
+
+    def accept(self):
+        try:
+            key = self.get_create_widget_text()
+            value = self.get_create_widget_value()
+            persist = self.get_persist_widget_value()
+            asyncio.ensure_future(self.dataset_ctl.set(key, pyon.decode(value), persist))
+            QtWidgets.QDialog.accept(self)
+        except:
+            logger.error("Cannot create dataset: "
+                         "input is not valid",)
+            return
+
+    def key_widget(self):
+        self.name_widget = QtWidgets.QLineEdit()
+        return self.name_widget
+
+    def edit_widget(self):
+        self.value_widget = QtWidgets.QLineEdit()
+        self.value_widget.setPlaceholderText('PYON (Python)')
+        return self.value_widget
+    
+    def persist_widget(self):
+        self.box_widget = QtWidgets.QCheckBox()
+        self.box_widget.setChecked(bool(False))
+        return self.box_widget
+
+    def get_create_widget_text(self):
+        return self.name_widget.text()
+
+    def get_create_widget_value(self):
+        return self.value_widget.text()
+    
+    def get_persist_widget_value(self):
+        return self.box_widget.isChecked()
```

Comment:
Those methods can be removed IMO. The ones that create the widgets are also not very well-written.